twine/package.py

```diff
@@ -174,10 +180,32 @@ def sign(self, sign_with, identity):
         if identity:
             gpg_args += ("--local-user", identity)
         gpg_args += ("-a", self.filename)
-        subprocess.check_call(gpg_args)
+        self.run_gpg(gpg_args)
 
         self.add_gpg_signature(self.signed_filename, self.signed_basefilename)
 
+    @classmethod
+    def run_gpg(cls, gpg_args):
+        try:
+            subprocess.check_call(gpg_args)
+            return
+        except FileNotFoundError:
+            print("{} executable not available.".format(gpg_args[0]))
+
+        if not gpg_args[0] == "gpg":
+            return
```

Comment:
This should be

```py
gpg_args[0] != "gpg"
```

And in this case, if we've tried with `gpg2` (which I presume is the assumption made here) we should *not* just silently allow a failure. People are expecting signatures to be generated, we must raise an exception and halt progress.

Reply:
Makes sense, I've updated the PR so that we always raise exception if not able to sign.